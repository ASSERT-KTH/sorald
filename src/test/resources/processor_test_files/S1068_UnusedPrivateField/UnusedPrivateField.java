import java.io.Serializable;

public class UnusedPrivateField implements Serializable {
    private int a = 42; // Noncompliant
    private static String b = "Hello world!"; // Noncompliant
    private static final long serialVersionUID = 42L; // Compliant as this field is used in deserialization
    protected double c = 3.14; // Compliant as this field may be used in subclasses
    private static int used = 1, unused = 2; // Compliant and noncompliant respectively

    public int compute(int number) {
        return number * used;
    }
}
