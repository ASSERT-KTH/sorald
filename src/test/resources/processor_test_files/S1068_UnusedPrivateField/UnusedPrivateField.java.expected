import java.io.Serializable;

public class UnusedPrivateField implements Serializable {
    private static final long serialVersionUID = 42L; // Compliant as this field is used in deserialization
    protected double c = 3.14; // Compliant as this field may be used in subclasses
    private static int d = 1; // Compliant as this field is used below

    public int compute(int number) {
        return number * d;
    }
}
